### PR TITLE
Coloured lighting & related stuff

### PR DESCRIPTION
**Pull request by TheGreatJorge**
- Makes all light tubes & bulbs have colour according to the light they emit. There are 6 preset colours: RGB CMY. Any other color is possible by simply changing the already present brightness_color variable. This PR makes it so it actually changes the color of the item and wall object too.
- Added two new colour defines used only along with preset coloured lights.
- Coloured lights are only obtainable by random maintenace loot, cargo spawn or trader.
- Added box of randomly mixed light colours and box for every preset light colour.
- Removed two sets of redundant defines.
- Converted light item New() to Initialize() (hopefully correctly).
- Made couple tiny tweaks to the lights code.

EDIT, 7. Oct 2017:
- When you hit light machinery or item with paint, it will change it's brightness_color var, instead of the color var.
- Paint buckets can now paint items and people again.
- You can now wash paint off by spraying yourself with fire extinguisher or otherwise covering yourself in water.

EDIT, 8. Oct 2017:
- Removed one apparently unused global list.
- Converted pain bucket New to Initialize and changed it's pathing style.
- Shower can now wash paint too.
- Fixed paint bucket sprites to be bit more dynamic and fixed the in hand sprite. Also added bucket lid.

Set of preset lights:
1113f069b65c947ee70c6fedd3401eea

Preset lights in action:
9f623dd4b4fa8eb30646b7a4dea5204f
5b0bfd8c49a836fed4332f3826789ce3
3fc051244d316373623617488725ba07
3cbff7963087b431352e0e103701f1ef
da11e5f1631fece139b89b8b8fdb7ad4
1ac108ecd0f344a04c012fe21838c077

Note that I am really not a colour expert so any feedback on these colours is welcome.
But also note that the item colour is same value as the actual emitted light colour, which might be a bit difficult to change.

### CODE REVIEW

**Comment by skull132** [comment]
Can you create these by dunking a light into a bucket of paint? Because that should also work!

**Comment by TheGreatJorge** [comment]
Oh, right. Nope. You can't. Only what is in the description. But I will see about making that work like that in next couple of days, now that you mention it.

**Comment by Lohikar** [comment]
How does this interact with nightmode's light fixture dim? Currently it just colors the fixture.

**Comment by TheGreatJorge** [comment]
I'll take a look at that too, to make sure it properly dims all lights. Currently it doesn't dim the texture, forgot about it.

**Comment by TheGreatJorge** [comment]
@Lohikar Apparently colouring the fixture while dimming is no problem. Doesn't seem to anyhow override the tube color. Just dims it.

**Comment by TheGreatJorge** [comment]
@skull132 I am not sure about the paint thing any more. Bucket to mob and object interactions seem to not work? Only paint bucket to turf seems to work. Not much I can do about that I think, if it was deliberately disabled. However, there seems to be a way around this, which might well be a bug. If you extract the paint with a dropper into beaker, you can splash and colour mobs and objects all fine. (Making new paint is also possible apparently, so I'd guess that is a way too?). So adding check for lights to colour them properly should not be hard I guess. But again, it might not work with buckets. And it seems the only clean way of doing it I think, because tying the code to buckets only at this point seems hacky to me.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm [hunk 1]
```diff
@@ -64,7 +64,21 @@
 		T.color = color
 
 /datum/reagent/paint/touch_obj(var/obj/O)
-	if(istype(O))
+	//special checks for special items
+	if(istype(O, /obj/item/weapon/reagent_containers))
+		return
```
Comment (TheGreatJorge) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm]: I thought it should check for containers first, and for technicolor stuff last, so I decided to separate those checks.

**Comment by TheGreatJorge** [comment]
@skull132 @Arrow768 The paint thing is in. Hopefully all is fine and well, should be ready for review.

File: code/game/objects/items/weapons/storage/boxes.dm [hunk 2]
```diff
@@ -773,6 +773,101 @@
 	for(var/i = 0; i < 7; i++)
 		new /obj/item/weapon/light/bulb(src)
 
+/obj/item/weapon/storage/box/lights/coloredmixed
+	name = "box of colored lights"
+	icon_state = "lightmixed"
+
+/obj/item/weapon/storage/box/lights/coloredmixed/fill()
+	..()
+	var/list/tube_colors = list(
```
Comment (skull132) [inline code/game/objects/items/weapons/storage/boxes.dm]: `var/static/list/tube_colors` please.

File: code/game/objects/items/weapons/storage/boxes.dm [hunk 3]
```diff
@@ -773,6 +773,101 @@
...
+	..()
+	var/list/tube_colors = list(
+		/obj/item/weapon/light/tube/colored/red,
+		/obj/item/weapon/light/tube/colored/green,
+		/obj/item/weapon/light/tube/colored/blue,
+		/obj/item/weapon/light/tube/colored/magenta,
+		/obj/item/weapon/light/tube/colored/yellow,
+		/obj/item/weapon/light/tube/colored/cyan
+	)
+	var/list/bulbs_colors = list(
```
Comment (skull132) [inline code/game/objects/items/weapons/storage/boxes.dm]: Also static.

File: code/controllers/subsystems/garbage-debug.dm [hunk 4]
```diff
@@ -646,7 +646,7 @@
 	SearchVar(default_medbay_channels)
 	SearchVar(rod_recipes)
 	SearchVar(NO_EMAG_ACT)
-	SearchVar(cached_icons)
+	SearchVar(last_chew)
```
[the remark this answers is not in the document]
Comment (TheGreatJorge) [reply code/controllers/subsystems/garbage-debug.dm]: Oh, this was merge conflict. I simply removed "SearchVar(cached_icons)", it was the unused list I found. And then manually resolved the conflict, I hope I did it correctly. "SearchVar(last_chew)" was already there.

[the remark this answers is not in the document]
Comment (TheGreatJorge) [reply code/controllers/subsystems/garbage-debug.dm]: As in: I removed "SearchVar(cached_icons)" line, comitted, it marked these two highlighted lines as merge conflict, so I resolved it by getting rid of the "SearchVar(cached_icons)" and leaving the "SearchVar(last_chew)" there. And it marked removal correctly, but marked me adding another line instead for some reason.

File: code/game/objects/items/weapons/paint.dm [hunk 5]
```diff
@@ -16,47 +14,68 @@ var/global/list/cached_icons = list()
...
-		paint_type = "black"
-
-	white
-		icon_state = "paint_white"
-		paint_type = "white"
+	var/paint_reagent = null //name of the reagent responsible for colouring the paint
+	var/paint_type = null //used for colouring detective technicolor coat and hat
+
+/obj/item/weapon/reagent_containers/glass/paint/Initialize()
+	..()
```
Comment (Lohikar) [inline code/game/objects/items/weapons/paint.dm]: `. = ..()`

File: code/modules/power/lighting.dm [hunk 6]
```diff
@@ -680,38 +741,46 @@
...
+	icon_state = "flight"
 	item_state = "egg4"
 	matter = list("glass" = 100)
 	brightness_range = 8
 	brightness_power = 0.8
+	randomize_range = FALSE
 
-// update the icon state and description of the light
+/obj/item/weapon/light/Initialize()
+	..()
```
Comment (skull132) [inline code/modules/power/lighting.dm]: also requires `. = ..()`

**Pull request merged**